toxcore/group.c

```diff
@@ -73,69 +129,47 @@ static int realloc_groupchats(Group_Chats *g_c, uint32_t num)
     return 0;
 }
 
+static void setup_conference(Group_c *g)
+{
+    memset(g, 0, sizeof(Group_c));
+    g->keep_join_index = -1;
+    g->live = true;
+}
 
 /* Create a new empty groupchat connection.
  *
  * return -1 on failure.
  * return groupnumber on success.
  */
-static int create_group_chat(Group_Chats *g_c)
+static aint create_group_chat(Group_Chats *g_c)
 {
-    uint32_t i;
+    aint i;
```

Comment:
It is save to compare 32/64-bit signed index variable with something 16-bit unsigned, because second op automatically cast to type of first op and full range of 16 bit unsigned can be placed into 32 bit signed without loss of bits. Even "i" will be negative, this comparison works correctly and without compiler warnings. Also, no need to force compiler use 16-bit variable for index. Compiler should use whole 32/64 bit register.